### PR TITLE
Support JSON_EXISTS, JSON_VALUE and JSON_QUERY functions

### PR DESCRIPTION
**Pull request by v-jizhang**
Cherry-pick of https://github.com/trinodb/trino/pull/9831/commits/ae0db1bb746254aa2ca6f5b0d90547d4fb4b08a0

Co-authored-by: kasiafi <30203062+kasiafi@users.noreply.github.com>

Test plan - Added tests

```
== RELEASE NOTES ==

General Changes
* Support JSON_EXISTS, JSON_VALUE and JSON_QUERY functions
```

### CODE REVIEW

File: presto-docs/src/main/sphinx/language/reserved.rst [hunk 1]
```diff
@@ -50,6 +50,9 @@ Keyword                        SQL:2016      SQL-92
 ``INTO``                       reserved      reserved
 ``IS``                         reserved      reserved
 ``JOIN``                       reserved      reserved
+``JSON_EXISTS``                reserved
```
Comment (mbasmanova) [inline presto-docs/src/main/sphinx/language/reserved.rst:53]: Would you add documentation for these new features?

Comment (v-jizhang) [reply presto-docs/src/main/sphinx/language/reserved.rst:53]: Document is in #18256.

**Review by steveburnett — CHANGES_REQUESTED** [review]
Local doc build fails with
"Running Sphinx v4.4.0
Sphinx version error: The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version. make: *** [html] Error 2"
Related to PR #21708.
Ran git merge master, then the doc build ran successfully.

Reviewed local doc build and the doc looks good. Update the PR and re-request my review.

**Comment by steveburnett** [comment]
Closing due to lack of activity and age. If this is still relevant, please re-open this PR, rebase, and resolve any file conflicts.

**Pull request closed without merging**